AWSIoTUnitTests/AWSIoTDataUnitTests.m

```diff
@@ -164,6 +164,30 @@ -(void)testRegisterIoTDataManagerWithConfigurationWithMQTTConfiguration {
 
 }
 
+-(void)testRegisterIoTDataManagerWithConfigurationUserNamePasswordWithMQTTConfiguration {
+    NSString *key = @"testRegisterIoTDataManagerWithMQTTConfigurationUserNamePassword";
+    AWSCognitoCredentialsProvider *credentialsProvider =
+    [[AWSCognitoCredentialsProvider alloc] initWithRegionType:AWSRegionUSWest2
```

Comment:
So far, all looks good. 
nit picking: Only here, since in https://docs.aws.amazon.com/iot/latest/developerguide/enhanced-custom-auth-using.html, it is mentioned is available only in the US East (N. Virginia) Region. Although the test is not actually talking to backend service. should we change it to `AWSRegionUSEast1`. 😂
```suggestion
    [[AWSCognitoCredentialsProvider alloc] initWithRegionType:AWSRegionUSWest2
```

Reply:
I would prefer not to make that distinction in unit tests. If we have a codepath that checks for existence of a beta endpoint, let's test it explicitly elsewhere, but otherwise, we should assume that unit tests are region-independent. If we use USWest2 elsewhere for whatever reason, we can continue to use that here.

Integ tests are a whole other story: be sure to set up resources for them through the integ test infrastructure scripts, as discussed with @wooj2.